src/magic_folder/test/test_web.py

```diff
@@ -710,3 +738,149 @@ def get_magic_folder(name):
+    service with Magic Folders of the given names.
+
+    :param unicode auth_token: The authorization token accepted by the
+        service.
+
+    :param [unicode] names: The names of the Magic Folders which will exist.
+
+    :return: An object like the ``treq`` module.
+    """
+    return treq_for_folders(auth_token, dict.fromkeys(names, {u"directory": None}))
+
+
+def treq_for_folders(auth_token, folders):
+    """
+    Construct a ``treq``-module-alike which is hooked up to a Magic Folder
+    service with Magic Folders like the ones given.
+
+    :param unicode auth_token: The authorization token accepted by the
+        service.
+
+    :param folders: A mapping from Magic Folder names to their configurations.
+        These are the folders which will appear to exist.
+
+    :return: An object like the ``treq`` module.
+    """
+    state = MagicFolderServiceState()
+    for name, config in folders.items():
+        state.add_magic_folder(name, config, object())
+
+    root = magic_folder_resource(state, lambda: auth_token)
+    return StubTreq(root)
+
+
+def magic_folder_config_for_local_directory(local_directory):
+    return {u"directory": local_directory}
+
+
+class ListMagicFolderTests(SyncTestCase):
+    """
+    Tests for listing Magic Folders using **GET /v1/magic-folder** and
+    ``V1MagicFolderAPI``.
+    """
+    url = DecodedURL.from_text(u"http://example.invalid./v1/magic-folder")
+    encoded_url = url_to_bytes(url)
+
+    @given(
+        tokens(),
+        sampled_from([b"PUT", b"POST", b"PATCH", b"DELETE", b"OPTIONS"]),
+    )
+    def test_method_not_allowed(self, auth_token, method):
+        """
+        A request to **/v1/magic-folder** with a method other than **GET**
+        receives a NOT ALLOWED or NOT IMPLEMENTED response.
+        """
+        treq = treq_for_folder_names(auth_token, [])
+        self.assertThat(
+            authorized_request(treq, auth_token, method, self.encoded_url),
+            succeeded(
+                matches_response(
+                    code_matcher=MatchesAny(
+                        Equals(NOT_ALLOWED),
+                        Equals(NOT_IMPLEMENTED),
+                    ),
+                ),
+            ),
+        )
+
+    @given(
+        tokens(),
+        dictionaries(
+            folder_names(),
+            absolute_paths(),
+        ),
+    )
+    def test_list_folders(self, auth_token, folders):
+        """
+        A request for **GET /v1/magic-folder** receives a response that is a
+        JSON-encoded list of Magic Folders.
+
+        :param dict[unicode, unicode] folders: A mapping from folder names to
+            local filesystem paths where we shall pretend the local filesystem
+            state for those folders resides.
+        """
+        treq = treq_for_folders(
+            auth_token, {
+                name: magic_folder_config_for_local_directory(path)
+                for (name, path)
```

Comment:
nit: shouldn't need the brackets here I think?

Reply:
True.  I find they help readability but I know that in other parts of the syntax, I find extraneous parens harm readability... It's pretty subjective I guess.  I think I'd like to leave them but I'm also up for adopting black (which seems to be the de facto formatting winner for Python) and making everyone do what it says (with CI/tool assistance).